ctapipe/io/eventsource.py

```diff
@@ -190,6 +190,18 @@ def is_stream(self):
         """
         return False
 
+    @property
+    @abstractmethod
+    def subarray(self):
```

Comment:
I don't know if that's a good idea. Are we we sure, subarray info can come from an event source? 
I don't think this is the case for e.g. LST real data event source

Reply:
In the current scheme the event source must provide it in one way or another (to attach it to the event container), even if the information is not located in the file. So I don't think this change should be a problem.

Reply:
ah ok

Reply:
yeah, the name EventSource may be confusing, but that's the idea - it provides the events + any other data associated with them (it already does this in fact! Just mixes the two concepts, so I don't see a problem)